merge/nested_test.go

```diff
@@ -500,6 +512,124 @@ func TestUpdateNestedType(t *testing.T) {
 				),
 			},
 		},
+		"struct_apply_remove_all": {
+			Ops: []Operation{
+				Apply{
+					Manager: "default",
+					Object: `
+						struct:
+						  name: a
+						  value: 1
+					`,
+					APIVersion: "v1",
+				},
+				Apply{
+					Manager: "default",
+					Object: `
+					`,
+					APIVersion: "v1",
+				},
+			},
+			Object: `
+			`,
+			APIVersion: "v1",
+			Managed:    fieldpath.ManagedFields{},
+		},
+		"struct_apply_remove_dangling": {
+			Ops: []Operation{
+				Apply{
+					Manager: "default",
+					Object: `
+						struct:
+						  name: a
+					`,
+					APIVersion: "v1",
+				},
+				Apply{
+					Manager: "default",
+					Object: `
+						struct:
+					`,
+					APIVersion: "v1",
+				},
+			},
+			Object: `
+				struct:
+			`,
+			APIVersion: "v1",
+			Managed: fieldpath.ManagedFields{
+				"default": fieldpath.NewVersionedSet(
+					_NS(
+						_P("struct"),
+					),
+					"v1",
+					true,
+				),
+			},
+		},
+		"struct_apply_update_remove_all": {
+			Ops: []Operation{
+				Apply{
+					Manager: "default",
+					Object: `
+						struct:
+						  name: a
+					`,
+					APIVersion: "v1",
+				},
+				Update{
+					Manager: "controller",
+					Object: `
+						struct:
+						  name: a
+						  value: 1
+					`,
+					APIVersion: "v1",
+				},
+				Apply{
+					Manager: "default",
+					Object: `
+					`,
+					APIVersion: "v1",
+				},
+			},
+			Object: `
+			`,
+			APIVersion: "v1",
+		},
+		"struct_apply_update_dict_dangling": {
+			Ops: []Operation{
+				Apply{
+					Manager: "default",
+					Object: `
+						struct:
+						  name: a
+					`,
+					APIVersion: "v1",
+				},
+				Update{
+					Manager: "controller",
+					Object: `
+						struct:
+						  name: a
+						  value: 1
+					`,
+					APIVersion: "v1",
+				},
+				Apply{
+					Manager: "default",
+					Object: `
+						struct: {}
+					`,
+					APIVersion: "v1",
+				},
+			},
+			Object: `
+				struct:
+				  value: 1
```

Comment:
This is also wrong, the user stopped expressing an opinion about `struct.name`. However, they co-owned that along with the controller. Since the controller still expresses an opinion, the field should remain.

Reply:
I take it back, the update shouldn't take ownership of the name field. so this seems right.

Reply:
I'll still add a test where update took over the name too and see what happens.